src/libstd/fs.rs

```diff
@@ -211,6 +211,14 @@ pub struct DirBuilder {
     recursive: bool,
 }
 
+/// How large a buffer to pre-allocate before reading the entire file at `path`.
+fn initial_buffer_size<P: AsRef<Path>>(path: P) -> usize {
+    // Allocate one extra byte so the buffer doesn't need to grow before the
+    // final `read` call at the end of the file.  Don't worry about `usize`
+    // overflow because reading will fail regardless in that case.
```

Comment:
Does "fail regardless" here mean `Result::Err` or panic?
`m.len() as usize + 1` can panic.

Reply:
It will panic in RawVec code when the vector's capacity exceeds `isize::MAX`.

Reply:
(or more likely, abort due to out-of-memory sometime before that)